pkg/httputils/httputils_test.go

```diff
@@ -0,0 +1,59 @@
+package httputils
+
+import (
+	"testing"
+)
+
+func TestDownload(t *testing.T) {
+	_, err := Download("http://docker.com")
+
+	if err != nil {
+		t.Errorf("Expected error to not exist when Download(http://docker.com)")
+	}
+
+	// Expected status code = 404
+	_, err = Download("http://docker.com/abc1234567")
+	if err == nil {
+		t.Errorf("Expected error to exist when Download(http://docker.com/abc1234567)")
+	}
+}
+
+func TestNewHTTPRequestError(t *testing.T) {
+	errorMessage := "Some error message"
+	httpResponse, _ := Download("http://docker.com")
+	err := NewHTTPRequestError(errorMessage, httpResponse)
+
```

Comment:
the test here is basically here as well https://github.com/Triara/docker/blob/increase-test-coverage/pkg/jsonmessage/jsonmessage_test.go#L14

you may instead check that by _Download_ ing some not found page `err.Code` is actually a 404 maybe..(not really useful tho)